user/tests/wiring/cell_connect_after_off/cellular_conn_after_off.cpp

```diff
@@ -0,0 +1,52 @@
+/**
+ ******************************************************************************
+  Copyright (c) 2015 Particle Industries, Inc.  All rights reserved.
+
+  This library is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation, either
+  version 3 of the License, or (at your option) any later version.
+
+  This library is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public
+  License along with this library; if not, see <http://www.gnu.org/licenses/>.
+ ******************************************************************************
+ */
+
+#include "application.h"
+#include "unit-test/unit-test.h"
+
+#if Wiring_Cellular == 1
+
+// Global variable to indicate a connection attempt
+volatile int g_state_conn_attempt = 0;
+
+static void nwstatus_callback_handler(system_event_t ev, int param) {
+    if (param == network_status_connecting){
+        g_state_conn_attempt = 1;
+    }
+}
+
+test(CELLULAR_CONN_01_conn-after-off)
```

Comment:
Nitpicking: prefer underscore in test names, as eventually it's just a function.